src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
Your version is shorter, but not very readable. It is better to announce changes for conditions separately.
An option is to define two variables isFemale and hasChildren, which contain filter functions for the corresponding conditions. Instead of using the conditions directly in the filter method, we use these variables to filter the people array.

<pre>
const isFemale = (person) => person.sex === 'f';
const hasChildren = (person) => people.some((p) => p.mother === person.name);

const women = withChildren
  ? people.filter((person) => isFemale(person) && hasChildren(person))
  : people.filter(isFemale);
  </pre>